### DISCUSSION TITLE
Task duration as a PlanningVariable

### DISCUSSION THREAD

**Discussion by slxnxl** [TimefoldAI/timefold-solver#72] [General]
In a task similar to the Food Packaging example, is it possible to calculate the duration of product packaging as a PlanningVariable?  Product packaging capacity per hour is known. Can timefold itself set the duration of packaging of a particular product on the line depending on the needs (for example for 1 hour or for 3 days and 5 hours)? Thank you!

**Comment by ge0ffrey** [comment]
Yes, through a shadow variable you can add properties that change as the assigned line changes:
https://timefold.ai/docs/timefold-solver/latest/shadow-variable/shadow-variable.html

Although you probably don't even need a shadow variable. Something like this might suffice:

    Duration getDuration() {
        return product.getDurationOnLine(line);
    }
